public/app/plugins/datasource/elasticsearch/configuration/DataLinks.test.tsx

```diff
@@ -1,84 +1,61 @@
+  it('should render correctly with no fields', async () => {
+    setup();
+
+    expect(screen.getByRole('heading', { name: 'Data links' }));
+    expect(screen.getByRole('button', { name: 'Add' })).toBeInTheDocument();
+    expect(await screen.findAllByRole('button')).toHaveLength(1);
   });
 
-  it('renders correctly when there are fields', async () => {
-    let wrapper: any;
-    await act(
-      // @ts-ignore we shouldn't use Promises in act => the "void | undefined" is here to forbid any sneaky "Promise" returns.
-      async () => {
-        wrapper = await mount(<DataLinks value={testValue} onChange={() => {}} />);
-      }
-    );
-
-    expect(wrapper.find(Button).filterWhere((button: any) => button.contains('Add')).length).toBe(1);
-    expect(wrapper.find(DataLink).length).toBe(2);
+  it('should render correctly when passed fields', async () => {
+    setup({ value: testValue });
+
+    expect(await screen.findAllByRole('button', { name: 'Remove field' })).toHaveLength(2);
+    expect(await screen.findAllByRole('checkbox', { name: 'Internal link' })).toHaveLength(2);
   });
 
-  it('adds new field', async () => {
+  it('should call onChange to add a new field when the add button is clicked', () => {
     const onChangeMock = jest.fn();
-    let wrapper: any;
-    await act(
-      // @ts-ignore we shouldn't use Promises in act => the "void | undefined" is here to forbid any sneaky "Promise" returns.
-      async () => {
-        wrapper = await mount(<DataLinks onChange={onChangeMock} />);
-      }
-    );
-    const addButton = wrapper.find(Button).filterWhere((button: any) => button.contains('Add'));
-    addButton.simulate('click');
-    expect(onChangeMock.mock.calls[0][0].length).toBe(1);
+    setup({ onChange: onChangeMock });
+
+    expect(onChangeMock).not.toHaveBeenCalled();
+    const addButton = screen.getByRole('button', { name: 'Add' });
+    fireEvent.click(addButton);
+
+    expect(onChangeMock).toHaveBeenCalled();
   });
 
-  it('removes field', async () => {
+  it('should call onChange to remove a field when the remove button is clicked', async () => {
     const onChangeMock = jest.fn();
-    let wrapper: any;
-    await act(
-      // @ts-ignore we shouldn't use Promises in act => the "void | undefined" is here to forbid any sneaky "Promise" returns.
-      async () => {
-        wrapper = await mount(<DataLinks value={testValue} onChange={onChangeMock} />);
-      }
-    );
-    const removeButton = wrapper.find(DataLink).at(0).find(Button);
-    removeButton.simulate('click');
-    const newValue = onChangeMock.mock.calls[0][0];
-    expect(newValue.length).toBe(1);
-    expect(newValue[0]).toMatchObject({
-      field: 'regex2',
-      url: 'localhost2',
-    });
+    setup({ value: testValue, onChange: onChangeMock });
+
+    expect(onChangeMock).not.toHaveBeenCalled();
+    const removeButton = await screen.findAllByRole('button', { name: 'Remove field' });
+    fireEvent.click(removeButton[0]);
```

Comment:
can we use `userEvent` here instead of `fireEvent`? test would need to be async then as well
```suggestion
    await userEvent.click(removeButton[0]);
```